e2e/docker_test.go

```diff
@@ -175,28 +185,47 @@ func setupPathDocker(
 		}()
 	}
 
-	// Handle termination signals
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	// Create a context that we can cancel
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// Set up signal handling
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt, syscall.SIGTERM)
+
+	// Create a channel to wait for cleanup completion
+	cleanupDone := make(chan struct{})
+
+	// Handle signals
 	go func() {
-		for sig := range c {
-			fmt.Printf("exit signal %d received\n", sig)
+		select {
+		case <-signalChan:
+			fmt.Println("\nReceived Ctrl+C, cleaning up containers...")
+			// Cancel the context
+			cancel()
+
+			// Perform cleanup
 			if err := pool.Purge(resource); err != nil {
-				fmt.Printf("could not purge resource: %s", err)
+				log.Printf("Could not purge resource: %s", err)
 			}
+
+			// Signal that cleanup is done
+			close(cleanupDone)
+		case <-ctx.Done():
+			// Context was canceled elsewhere
 		}
 	}()
 
 	if err := resource.Expire(containerExpirySeconds); err != nil {
 		t.Fatalf("[ERROR] Failed to set expiration on docker container: %v", err)
 	}
 
-	fmt.Println("  ✅ PATH test container started successfully!")
-	fmt.Println("🏥 Performing health check on PATH test container...")
+	t.Logf("  ✅ PATH test container started successfully!")
```

Comment:
My preference is definitely to use `fmt.` rather than `t.Logf` as it results in extra noise like logging the file and line which doesn't add a lot of value to the output IMO.